### ISSUE TITLE
Hello & 500 Error on Signup

### ISSUE THREAD

**Issue by colinwd** [cmazuc/scarfage#62]
Hi guys! My name's Colin, I'm a software dev and Jason's recruited me to help out with the cause. @cmazuc Jason tells me you're the go to guy for how everything works. Looking forward to helping out any way I can!

I've also got a bug report! Think it's possibly my password that caused an issue - 30 characters long with punctuation. However the registration appears to have still worked normally - I received a signup email and can log in to my account without any trouble.

What is your username?
colinwd

Did you receive a specific error message?
500 Error on user creation

What was the URL that gave an error?
Hitting the `Submit` button on https://www.scarfage.com/newuser

Approximately what time did the error occur?
4:50 PM PST

**Comment by cmazuc** [comment]
Thanks for the bug report and looking forward to having some more help on this :)

Unfortunately I was only able to reproduce this once on the main site and haven't been and to repro at all on test or locally. I'm thinking this may have been a db issue but I'm not entirely sure as there wasn't an error logged beyond the generic 500. I did find some issues with how whitespace was handled but you would have received a slightly different error.

Shot in the dark: did the welcome email have your IP listed or was it missing?

**Comment by colinwd** [comment]
You're in luck!

`2016-02-28 00:51:34. New user request was received from 73.181.210.6`

**Comment by colinwd** [comment]
I'll be working on getting my dev environment set up tonight - I'll see if I can reproduce with my specific password once I get up and running.

Other than that, is there a general list of priorities I can check out? My own inclination is to start out with some unit tests as that always helps me dig in and understand a codebase.

**Comment by cmazuc** [comment]
I sent this separately but I'm not sure where it landed... Reposting with a few edits.

Unit tests are definitely an area in which we are lacking. :)

So I'm not super happy with the organization right now, I'd like to break up scarflib.py and scrub the remaining sql from other parts of the project. I'd also like to remove the Flask stuff from scarflib. Imports are a bit of a mess. Auth checks kinda suck, I was going to redo it with decorators [edit: already done, not committed to master yet]. There's a lot of other little cleanup that I'd like to do but I don't have a list. I'd be curious to know your thoughts on what's here so far, I'm not really a dev and hadn't written a line of python before starting.

I'm currently working on the item tagging stuff, it's a bit of a mess right now and to be honest I'm not sure how it'll work once the database grows. Queries are quick but I don't think it'll perform well on updates once the tree gets big. If you want to take that on then it's all yours.

Jason would love it if we had facebook auth integration. There's a ton of stuff that needs to be fixed with the frontend, I haven't put much more into it then I needed to verify functionality [edit: unit tests...]. Other than that we've been trying to collect everything on github issues. Jason has a bunch more stuff but it hasn't made it in [here] yet.

I went to add some unit tests to scarflib a while back and found that all the Flask stuff make this fairly difficult and realized at that point that I should have done it a bit differently. Given that this is essentially a hobby project for me (and one that takes up nearly all of my hobby time) it wasn't a high priority. Let me know if you already worked around this or if it's blocking your testing.

**Comment by cmazuc** [comment]
scarflib has been broken up into a proper lib with no unnecessary dependencies on flask, should make testing a bit easier. Let me know if you already have something in progress

**Comment by cmazuc** [comment]
I've added some better logging around new user signup, please let me know if you can repro

**Comment by cmazuc** [comment]
Can't repro, closing for now
